contrib/http_util/src/main/java/io/opencensus/contrib/http/HttpClientHandler.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.contrib.http;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import io.opencensus.trace.MessageEvent.Type;
+import io.opencensus.trace.Span;
+import io.opencensus.trace.SpanContext;
+import io.opencensus.trace.Tracer;
+import io.opencensus.trace.propagation.TextFormat;
+import javax.annotation.Nullable;
+
+/**
+ * This helper class provides APIs for instrumenting HTTP clients.
+ *
+ * @since 0.13
+ */
+public class HttpClientHandler<Q, P> extends HttpHandler<Q, P> {
```

Comment:
Should this class be final?

Reply:
No. Some of the methods are intended to be overridden by user if they want their own implementations.